pkg/datapath/linux/node.go

```diff
@@ -522,6 +525,55 @@ func (n *linuxNodeHandler) encryptNode(newNode *node.Node) {
 
 }
 
+func neighborLog(spec string, err error, ip *net.IP, hwAddr *net.HardwareAddr, link int) {
+	scopedLog := log.WithFields(logrus.Fields{
+		logfields.Reason: spec,
+		"IP":             ip,
+		"HardwareAddr":   hwAddr,
+		"LinkIndex":      link,
+	})
+
+	if err != nil {
+		scopedLog.WithError(err).Error("insertNeighbor failed")
+	} else {
+		scopedLog.Info("insertNeighbor")
+	}
+}
+
+func (n *linuxNodeHandler) insertNeighbor(newNode *node.Node) {
+	if newNode.IsLocal() {
+		return
+	}
+
+	ciliumIPv4 := newNode.GetNodeIP(false)
+	var hwAddr net.HardwareAddr
+	link := 0
+
+	if option.Config.Tunnel != option.TunnelDisabled {
+		return
+	}
+
+	linkAttr, err := netlink.LinkByName(option.Config.EncryptInterface)
+	if err != nil {
+		neighborLog("insertNeightbor LinkByName", err, &ciliumIPv4, &hwAddr, link)
+		return
+	}
+	link = linkAttr.Attrs().Index
+
+	if hwAddr, _, err := arping.Ping(ciliumIPv4); err == nil {
```

Comment:
From inspection, I note that this may block for up to 500ms. Is this likely to cause any hiccups in this goroutine?